### ISSUE TITLE
Replace our forms with Mui forms

### ISSUE THREAD

**Issue by LeandraH** [digitalfabrik/integreat-app#3365]
### Describe the Problem

Mui has a very nice form system that we should start using.

One important difference: we have so far disabled the submit button when required fields were still missing. This is not best practice for A11y, the submit button should not be disabled but a useful error message appear if the button is pressed before all required fields are filled in. Mui probably does that out of the box.

### Describe Your Preferred Solution

- Replace our forms with Mui ones
- Make sure that submit buttons are enabled even when required fields are empty, but that a user that presses the submit button too early is informed of what is missing.

### Related Issues

Part of: #3231 

### Additional Information

Work has already started here: #3365

https://mui.com/material-ui/api/form-control/

https://mui.com/material-ui/api/form-group/

**Comment by steffenkleinle** [comment]
JFYI: This might heavily conflict with #3366, so beware of merge conflicts (or even wait until it is merged).
